### ISSUE TITLE
coqide should come with a coqide.desktop file

### ISSUE THREAD

**Issue by SnarkBoojum** [coq/coq#15182]
Most desktop-oriented apps come with a .desktop file, which gets shipped in /usr/share/applications/ on Debian. You can read about them [here](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html).

In coq 8.14.0, there is no coqide.desktop, but there is one in the Debian package, written by Ralf Treinen ; here is its content:
```
[Desktop Entry]
Name=CoqIDE Proof Assistant
Comment=Graphical interface for the Coq proof assistant
Exec=coqide
Type=Application
Categories=Development;Science;Math;IDE;GTK;
Terminal=false
Icon=coq
Keywords=Science;Math
```

you might want to include it yourselves, and improve on it (adding translations, for example).

**Comment by Alizter** [comment]
Why exactly do we want to include one here if the debian package already has one? The `.desktop` file seems to be only relevent for a few select Linux distributions so it seems pretty irrelevant elsewhere. Not to mention, coq (together with coqide) is typically installed using opam. What use is the desktop file there?

**Comment by SnarkBoojum** [comment]
You seem highly confused:
- Linux distributions and the *BSD are first-class distributions, providing complete operating systems, running computers from the bare metal up to the top, and making a ton of software available to all users on the system ;
- opam is as far as I know a second-class distribution, providing only little software, for a single user on each system and depends on a real distribution to get it up and staggering.

"Both" (the first item is a big plenty) allow users to **install** coqide.

But once it's there -- no matter how, as a graphical application, it's nice to have it available in whatever menu the user is running, ready to be directly launched or added in a quick-launch series of icons for later everyday use. This is about being user-friendly.

And there, on linux distributions and the *BSD (which many of coq users are running), many types of menus are available, depending on the user interface chosen (gnome, kde, xfce, lxde, ... again a long list). But there is a good news: the developers of all these desktop systems got together years ago (twenty?) and agreed on a common way to declare graphical applications so they appear in their specific menu.

This is what .desktop files are about. You already ship an icon. With a .desktop file, on a same system, once the software is installed, even if the different users use different graphical environments, the system-wide installation of coqide is available to all of them in an easy-to-go way. Opam could install it too, as there are places where to put a .desktop for the software installed by a single user in her or his home directory.

In Debian, since no .desktop was available upstream, Ralf Treinen wrote one. [Here](https://src.fedoraproject.org/rpms/coq/blob/rawhide/f/fr.inria.coqide.desktop) is what they have in Fedora, and I'm pretty sure other distributions have one too. But it's always only rudimentary. By making it common, the coq developers can make it better.

**Comment by Zimmi48** [comment]
Indeed, desktop files are useful for a wide variety of distributions (nixpkgs / NixOS got its own `coqide.desktop` file recently: https://github.com/NixOS/nixpkgs/pull/138072). I'd be surprised if it was not possible to install software with desktop files with opam as well.

Without a desktop file, you can only launch CoqIDE through the console. It can be useful to quickly launch CoqIDE directly from the desktop for quick uses (even though you probably still want to launch it from the console when working on an actual project).
